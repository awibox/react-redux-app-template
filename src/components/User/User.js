import React from 'react';
import PropTypes from 'prop-types';
// Styles
import styles from './User.scss';

const User = (props) => (
  <div className={styles.user}>
    <i className="fa fa-user"/>
    {props.children}
  </div>
);

User.propTypes = {
  children: PropTypes.any,
};

export default User;
