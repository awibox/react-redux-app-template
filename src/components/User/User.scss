@import '../../styles/variables';

.user {
  padding: 5px 0;
  i {
    margin-right: 5px;
  }
}
