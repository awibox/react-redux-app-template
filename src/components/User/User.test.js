import React from 'react';
import User from './User';
import { shallow } from 'enzyme';

describe('User', () => {
  it('should render correctly', () => {
    const component = shallow(<User>Test user</User>);
    expect(component).toMatchSnapshot();
  });
});
