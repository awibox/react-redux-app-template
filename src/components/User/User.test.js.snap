// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render correctly 1`] = `
<div
  className="user"
>
  <i
    className="fa fa-user"
  />
  Test user
</div>
`;
